### ISSUE TITLE
How to rerender on resize and how to configure jest properly to use with expo ?

### ISSUE THREAD

**Issue by kopax** [expo/match-media#5]
Hello, I have just read this recent article and tried their sample to test your module https://medium.com/p/e0b73ed5777b/responses/show

```jsx
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import '@expo/match-media'
import { useMediaQuery } from 'react-responsive';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#5be54b',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

const stylesDesktop = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#327ae5',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
let i = 0;
export default function App() {
  const isTabletOrMobileDevice = useMediaQuery({
    maxDeviceWidth: 100,
  });
  console.log('rendering', ++i);
  if (isTabletOrMobileDevice) {
    return (
      <View style={styles.container}>
        <Text>Hi Mobile Users</Text>
      </View>
    );
  }
  return (
    <View style={stylesDesktop.container}>
      <Text>Hello Desktop people!</Text>
    </View>
  );
}
```

This produce an error when using with the default jest configuration: 

```js
    TypeError: Cannot read property 'Orientation' of undefined

      at new MediaQuery (node_modules/@expo/match-media/src/polyfill.native.ts:12:23)
      at call (node_modules/@expo/match-media/src/polyfill.native.ts:69:43)
      at new Mql (node_modules/react-responsive/dist/webpack:/dist/react-responsive.js:337:28)
      at __WEBPACK_IMPORTED_MODULE_1_matchmediaquery___default (node_modules/react-responsive/dist/webpack:/dist/react-responsive.js:377:10)
      at getMatchMedia (node_modules/react-responsive/dist/webpack:/dist/react-responsive.js:171:12)
      at mountState (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:6520:20)
      at Object.useState (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:7077:16)
      at Object.useState (node_modules/react/cjs/react.development.js:1562:21)
      at useMatchMedia (node_modules/react-responsive/dist/webpack:/dist/react-responsive.js:174:72)
      at useMediaQuery (node_modules/react-responsive/dist/webpack:/dist/react-responsive.js:217:12)

  console.error node_modules/react-test-renderer/cjs/react-test-renderer.development.js:11958
    The above error occurred in the <App> component:
        in App
    
    Consider adding an error boundary to your tree to customize error handling behavior.
    Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
```

Also, when I try to resize, it does not rerender, I was not able to get a mobile view in web.

- Is this expected? 
- How can I configure jest in order to work with unit testing?

Thanks for sharing and best!

**Comment by kopax** [comment]
Error is different when using `0.0.0-alpha.2`:

```
TypeError: mappedCoverage.addStatement is not a function
    at /home/dka/workspace/git.kopaxgroup.com/momagroup/icimatin-native/node_modules/istanbul-lib-source-maps/lib/transformer.js:30:32
    at Array.forEach (<anonymous>)
    at SourceMapTransformer.processFile (/home/dka/workspace/git.kopaxgroup.com/momagroup/icimatin-native/node_modules/istanbul-lib-source-maps/lib/transformer.js:22:38)
    at /home/dka/workspace/git.kopaxgroup.com/momagroup/icimatin-native/node_modules/istanbul-lib-source-maps/lib/transformer.js:120:34
    at Array.forEach (<anonymous>)
    at SourceMapTransformer.transform (/home/dka/workspace/git.kopaxgroup.com/momagroup/icimatin-native/node_modules/istanbul-lib-source-maps/lib/transformer.js:109:29)
    at MapStore.transformCoverage (/home/dka/workspace/git.kopaxgroup.com/momagroup/icimatin-native/node_modules/istanbul-lib-source-maps/lib/map-store.js:173:14)
    at /home/dka/workspace/git.kopaxgroup.com/momagroup/icimatin-native/node_modules/@jest/reporters/build/coverage_reporter.js:225:59
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/home/dka/workspace/git.kopaxgroup.com/momagroup/icimatin-native/node_modules/@jest/reporters/build/coverage_reporter.js:124:24)
```

**Comment by listiani13** [comment]
TLDR: Using `maxWidth` should work

Hey! I notice that you used `maxDeviceWidth` instead of `maxWidth`. `maxDeviceWidth` refers to the actual device's size, so it won't change the layout if we resize our browser. However, if you use & resize in Responsive Design Mode (in Firefox/Chrome), it'll change because the width change is applied as the device width. Was running into this thing too 😂

References:
https://developer.mozilla.org/en-US/docs/Web/CSS/@media/device-width
https://stackoverflow.com/questions/6747242/what-is-the-difference-between-max-device-width-and-max-width-for-mobile-web
